pkg/client/client.go

```diff
@@ -62,7 +62,17 @@ func NewClient(token, apiEndpoint string) (runtimeclient.Client, error) {
 }
 
 func NewClientFromRestConfig(config *rest.Config) (runtimeclient.Client, error) {
-	config.Insecure = true
+	tc, err := config.TransportConfig()
+	if err != nil {
+		return nil, err
+	}
+	// when specifying the insecure flag with a certificate file the runtime client throws the following error:
+	// 	cannot create client: specifying a root certificates file with the insecure flag is not allowed
+	//
+	// set insecure flag only if certificate file is not present
+	if !tc.HasCA() {
+		config.Insecure = true
```

Comment:
It seems to me that this change could potentially create problems in non-CI/dev environments. Indeed, if a kubeconfig is passed where the CA is mistakenly not set, it will assume by itself to use Insecure connection. When connecting to a production cluster, I'd say we should always use Secure connection and fail if the Kubeconfig is not explicitly requiring Insecure connection.

Reply:
I see your point, but apparently right now (without this change) we always use insecure connection for the rest client, unless I'm missing something. I don't remember exactly why we had to force the insecure flag here, but I agree that it should be driven by the kubeconfig probably.

Reply:
in this particular case, we probably don't need any special handling at all. This function is used to create a client from kubeconfig which should contain all data/information we need. 
There is another function that creates a client based on the SA tokens from sandbox.yaml files which doesn't have any CA set at all. To make our lives easier we simply set the insecure flag, but it's a technical dept which should be fixed.

Reply:
Yeah I think we can rely on what is in the kubeconfig, since the current implementation fails when the cluster has a trusted CA apparently.